web/src/search_suggestion.ts

```diff
@@ -668,6 +668,37 @@ function get_public_channels_filter_suggestions(
     ];
     return get_special_filter_suggestions(last, terms, suggestions);
 }
+
+function get_archived_channel_filter_suggestions(
+    last: NarrowTerm,
+    terms: NarrowTerm[],
+): Suggestion[] {
+    let search_string = "channels:archived";
+    // show "channels:archived" option for users who
+    // have "streams" in their muscle memory
+    if (last.operator === "search" && common.phrase_match(last.operand, "streams")) {
+        search_string = "streams:archived";
+    }
+    let description_html = Filter.describe_public_channels(last.negated ?? false);
+    description_html = description_html.charAt(0).toUpperCase() + description_html.slice(1);
+    const suggestions: SuggestionAndIncompatiblePatterns[] = [
+        {
+            search_string,
+            description_html,
+            is_people: false,
+            incompatible_patterns: [
+                {operator: "is", operand: "dm"},
+                {operator: "channel"},
+                {operator: "dm-including"},
+                {operator: "dm"},
+                {operator: "in"},
+                {operator: "channels"},
```

Comment:
This isn't incompatible, you can have `channels:public` `channels:archived`. I don't think `in:` is incompatible either.